lib/IRGen/IRGenModule.cpp

```diff
@@ -693,6 +693,7 @@ void IRGenModule::emitAutolinkInfo() {
     }
     break;
   }
+  case llvm::Triple::COFF:  // for cygwin
```

Comment:
The cygwin tool chain covers the long section names and generates long section name contained binaries. The COFF format has a separated string table for names longer than 8, but the table is dropped when loaded to system memory. At run-time, the long name section is exist in memory but can not be identified.

Without 'swift2_protocol_conformances', print("Hello") didn't work, so I had to modify it.
But I don't know that how 'swift1_autolink_entries' works and didn't touch it.

I think that If it must be accessed at run-time, it also has to have a shorter name.
But I hope keep it unchanged now and resolve it in another commit later.